osprey_flask_app/utils.py

```diff
@@ -21,105 +22,152 @@ def get_filename_from_path(path):
+    Parameters
+        1. arg_dict (dict): dictionary containing coordinates and mapping to pour points file
+    """
+    lons = arg_dict["lons"].split(",")
+    lats = arg_dict["lats"].split(",")
+    names = arg_dict["names"].split(",")
+    try:
+        long_names = arg_dict["long_names"].split(",")
+        pour_points = "lons,lats,names,long_names\n"
+        pour_points += "".join(
+            [
+                ",".join((str(lon), str(lat), str(name), str(long_name))) + "\n"
+                for (lon, lat, name, long_name) in zip_longest(
+                    lons, lats, names, long_names
+                )  # Coordinates with missing attributes are padded with 'None' values
+            ]
+        )
+    except AttributeError:  # long_names not given
+        pour_points = "lons,lats,names\n"
+        pour_points += "".join(
+            [
+                ",".join((str(lon), str(lat), str(name))) + "\n"
+                for (lon, lat, name) in zip_longest(lons, lats, names)
+            ]
+        )
+    arg_dict["pour_points"] = pour_points[:-1]  # Remove last new line character
+
+
 def create_full_arg_dict(args):
     """Create full dictionary of arguments from request url to pass to osprey.
     Add 'None' values for missing arguments.
     Parameters
         1. args (request.args): arguments given by url
     """
 
-    # Expected url arguments (format is <arg:default_value>)
-    exp_args = [
-        "case_id",
-        "grid_id",
-        "run_startdate",
-        "stop_date",
-        "pour_points",
-        "uh_box",
-        "routing",
-        "domain",
-        "input_forcings",
-        "params_config_file",
-        "params_config_dict",
-        "convolve_config_file",
-        "convolve_config_dict",
+    # Optional url arguments (format is <arg:default_value>)
+    opt_args = [
+        "long_names:None",
+        "params_config_dict:None",
+        "convolve_config_dict:None",
         "version:1",
         "loglevel:INFO",
         "np:1",
     ]
-    arg_dict = {}
-    for arg in exp_args:
-        if ":" not in arg:
-            arg_dict[arg] = args.get(arg)
-        else:
-            (arg, default) = arg.split(":")
-
-            # Convert default value to int if possible
-            try:
-                default = int(default)
-            except ValueError:
-                pass
-
-            arg_dict[arg] = args.get(arg, default=default)
+    arg_dict = dict(args)
+    for arg in opt_args:
+        (arg, default) = arg.split(":")
+        if default == "None":
+            default = None
+
+        # Convert default value to int if possible
+        try:
+            default = int(default)
+        except (ValueError, TypeError):
+            pass
+
+        arg_dict[arg] = args.get(arg, default=default)
+
+    # Obtain proper input files from THREDDS
+    get_input_files(arg_dict)
+
+    # Concatenate lons, lats, and names into long pour points string
+    create_pour_points(arg_dict)
```

Comment:
These comments are a bit redundant since the method names are nice and descriptive. Plus you have nice docstrings for people to read if they need more info.